server/src/main/java/org/elasticsearch/index/mapper/vectors/DenseVectorFieldMapper.java

```diff
@@ -2088,16 +2110,34 @@ && isNotUnitVector(squaredMagnitude)) {
                     }
                 }
             }
+
+            int adjustedNumCands = numCands;
+            if (needsRescore(numCandsFactor)) {
+                // We shouldn't have less than k candidates (or 1 in case k is not set) to rescore
+                int minCands = k == null ? 1 : k;
+                // k <= numCands * numCandsFactor <= NUM_CANDS_OVERSAMPLE_LIMIT. Adjust otherwise.
+                adjustedNumCands = Math.max(minCands, (int) Math.ceil(numCands * numCandsFactor));
+                adjustedNumCands = Math.min(adjustedNumCands, NUM_CANDS_OVERSAMPLE_LIMIT);
+            }
```

Comment:
This logic doesn't make sense if we are allowing `< 1.0` resample factors. I am thinking we shouldn't allow less than 1.0. 

What was the plan for allowing < 1.0?

Reply:
I was thinking on allowing to rescore less than `num_candidates`, but more than `k` (like the previous iteration).

For num_cands = 100, numCandsFactor = 0.2, k = 10, I was thinking it would explore 100 num_candidates, but rescore just the top 20, and return the top rescored 10 for each shard. Does it make sense?

Reply:
I understand, but the way this is implemented doesn't seem to actually be that? You are passing the adjusted candidates down to the queries. So, they will gather fewer candidates & only rescore those?

Some testing for this would be good if this is what we are wanting.

Reply:
You are correct, I will need to change the implementation to allow for that - I was incorreclty limiting the results on the query, and also the num_candidates as you mention.

I see now that the proposal above is not very consistent, as having factor > 1.0 would expand the number of candidates, but making it < 1.0 would keep them and instead affect the rescoring. That's confusing.

I think it will be easier and less confusing to just allow for factor >= 1.0. Users can decide to have less num_candidates in case they want to rescore less results instead of applying a factor.

I will make the test changes and also simplify the implementation of the rescore query given that we don't have to modify `k` anymore.

Reply:
> I think it will be easier and less confusing to just allow for factor >= 1.0. Users can decide to have less num_candidates in case they want to rescore less results instead of applying a factor.

++

sounds good